CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+
+Once your PR is submitted, it will be reviewed by the maintainers. We may suggest changes or ask for clarifications before merging.
+
+#### IMPORTANT NOTE
+
+Please don't force push commits in your branch, in order to keep commit history and make it easier for us to see changes between reviews.
+
+Make sure to Allow edits of maintainers (under the text box) in the PR so people can actually collaborate on things or fix smaller issues themselves.
+
+#### Rebasing and other workflows
+
+(taken from: [openage on rebasing](https://github.com/SFTtech/openage/blob/master/doc/contributing.md#rebasing))
+
+**Rebasing** is 'moving' your commits to a different parent commit.
+
+In other words: *Cut off* your branch from its tree, and *attach it* somewhere else.
+
+There's two main applications:
+
+- If you based your work on a older master (so old that stuff can't be automatically merged),
+  you can rebase to move your commits to the current [upstream](https://help.github.com/articles/fork-a-repo/) master:
+
+```bash
+# update the upstream remote to receive new commits
+git fetch upstream
+
+# be on your feature branch (you probably are)
+git checkout my-awesome-feature
+
+# make backup (you never know, you know?)
+git branch my-awesome-feature-backup
+
+# rebase: put your commits on top of upstream's master
+git rebase -m upstream/master
+```
+
+- If you want to fix an older commit of yours, or merge several commits into a single one (**squash** them), rebase interactively.
+  We ***don't*** want to have a commit history like this:
+
+  - `add stuff`
+  - `fix typo in stuff`
+  - `fix compilation`
+  - `change stuff a bit`
+  - and so on...
+
+##### `rebase` in practice
+
+`git log --graph --oneline` shows your commit history as graph.
+To make some changes in that graph, you do an **interactive rebase**:
+
+```
+git rebase -i -m upstream/master
+```
+
+With this command, your new "base" is `upstream/master` and you can
+then change any of your branch's commits.
+
+`-i` will open an interactive editor where you can choose actions for each individual commit:
+
+- re-order commits
+- drop commits by deleting their line
+- squash/fixup ("meld") your commits
+- reword a commit message
+- stop rebasing at a commit to edit (`--amend`) it manually
+
+Just follow the messages on screen.
+
+##### Changing commits with `amend` and `fixup`
+
+There's also `git commit --amend` which is a "mini-rebase" that modifies just the last commit with your current changes by `git add`.
+It just skips the creation of a new commit and instead melds the changes into the last one you made.
+
+If you want to update a single commit in the range `[upstream/master, current HEAD]` which is not the last commit:
+
+- `edit stuff you wanna change in some previous commit`
+- `git add changed_stuff`
+- `git commit --fixup $hash_of_commit_to_be_fixed`
+- `git rebase --autosquash -i -m upstream/master`
```

Comment:
Wow cool - didn't know about this possibility! Have to test it out! I so far used `git rebase -i` and manually resorted the fixups... Thanks for that hint!